fairlearn/postprocessing/_threshold_optimizer.py

```diff
@@ -267,7 +275,7 @@ def _threshold_optimization_demographic_parity(sensitive_features, labels, score
             selection_error_curve[sensitive_feature_value]['error']
 
         logger.debug(OUTPUT_SEPARATOR)
-        logger.debug("Processing " + str(sensitive_feature_value))
+        logger.debug("Processing {0}", str(sensitive_feature_value))
```

Comment:
why suddenly {0} here? Above it looks different, just
logger.debug("some string ", value)

Which one is right?

Reply:
if str(...) is required then it's required above as well

Reply:
It's one where there was a complaint about not deferring the formatting of the log string. The basic idea being that it's best to wait until you're sure that a log message is going to be written to interpolate the string. I do recognise that the `str(..)` call does go somewhat against that....